beacon-chain/db/kv/kv.go

```diff
@@ -243,10 +243,38 @@ func (s *Store) setupBlockStorageType(isFreshDatabase bool) error {
 			return bkt.Put(saveBlindedBeaconBlocksKey, []byte{1})
 		})
 	}
-	saveBlinded, err := s.shouldSaveBlinded()
-	if err != nil {
+
+	// We check if we want to save blinded beacon blocks by checking a key in the db
+	// otherwise, we check the last stored block and set that key in the DB if it is blinded.
+	var saveBlinded bool
+	if err := s.db.Update(func(tx *bolt.Tx) error {
```

Comment:
Wouldn't it be simpler to just read the head block and then check that ? We should always have a head root saved for non-fresh databases. 

Also the cursor search logic is unsound, keys in bolt are byte-sorted. So If you want the last block inserted into the db, you will need to do the highest slot search. The cursor seeking the last key in the bucket doesn't really work because we insert `[block_root, ssz_marshalled_block]`. Block roots are random, so it would just have the last byte-sorted key in the bucket rather than the last inserted block.